include/core/Partition_Parser.hpp

```diff
@@ -0,0 +1,168 @@
+            boost::property_tree::json_parser::read_json(file_path, loaded_tree);
+            this->tree = loaded_tree;
+            std::cout << "file_path: " << file_path << std::endl;
+        };
+
+        virtual ~Partitions_Parser(){};
+
+        //The function that parses the json file and build a unordered map of structs for each line in the json list
+        void parse_partition_file() {
+            std::cout << "\nroot_tree: " << tree.size() << std::endl;
+
+            //Declare two partition_data type structs
+            PartitionData part_data;
+
+            //Declare temporary variables
+            std::vector<std::string> remote_nexi_tmp;
+            std::vector<std::string> remote_catchments_tmp;
+            std::vector<int> remote_mpi_ranks_tmp;
+
+            //The outter loop iterating through the list of partitions
+            int part_counter = 0;
+            for(auto &partition: tree.get_child("partitions"))  {
+                std::string part_id = (partition.second).get<std::string>("id");
+
+                //Get mpi_world_rank and set the corresponding part_data struct member
+                int mpi_rank = std::stoi(part_id);
+                part_data.mpi_world_rank = mpi_rank;
+
+                geojson::JSONProperty part = geojson::JSONProperty(part_id, partition.second);
+
+                //Get cat_ids list and set the corresponding part_data struct member
+                for (auto &remote : part.at("cat-ids").as_list())
+                {
+                    cat_ids.push_back(remote.as_string());
+                }
+                part_data.cat_ids = cat_ids;
+                cat_ids.clear();
+
+                //Get remote-connections and set the corresponding part_data struct member
+                for (auto &remote_map : part.at("remote-connections").as_list())
+                {
+                    std::string remote_nex_id = remote_map.at("nex-id").as_string();
+                    std::string remote_cat_id = remote_map.at("cat-id").as_string();
+                    int remote_mpi_rank = remote_map.at("mpi-rank").as_natural_number();
+                    remote_nexi_tmp.push_back(remote_nex_id);
+                    remote_catchments_tmp.push_back(remote_cat_id);
+                    remote_mpi_ranks_tmp.push_back(remote_mpi_rank);
+                }
+                part_data.remote_nexi = remote_nexi_tmp;
+                part_data.remote_catchments = remote_catchments_tmp;
+                part_data.remote_mpi_ranks = remote_mpi_ranks_tmp;
+                remote_nexi_tmp.clear();
+                remote_catchments_tmp.clear();
+                remote_mpi_ranks_tmp.clear();
+
+                //Insert part_data struct into unordered map based on part_id
+                partition_ranks.emplace(part_id, part_data);
+
+                part_counter++;       
+            }
+            
+            //return;
+        };
+
+        typedef PartitionData part_strt;
```

Comment:
I find this typedef usage a bit confusing.  The alias is a little ambiguous to me, though that by itself I probably wouldn't have brought up.  But just a few lines later, `PartitionData` is explicitly used instead of the alias.  The only place the alias is used is on the very next line for a return type.

Reply:
Just wanted to use a different name for the function type definition and the variable name inside the function to reduce confusion. Maybe it had the unintended opposite effect.

Reply:
The array are not needed to be the same size. 
Thanks for the careful check on the variable nex-ids. Indeed it currently used anywhere in the code. I did not delete it from the code thinking just in case we might need local nexuses in the future. This can be changed.

Reply:
Sorry, I meant nex-ids is currently not used anywhere in the code.

Reply:
Well, if it was done deliberately for purposes of code style, I don't want to unilaterally dictate that this is "bad."  It just looked like it might have actually been unintentional: e.g., an approach using the typedef that was started, but then abandoned, and not fully cleaned up.

We do need to discuss and adopt some code style conventions, but until we have those, I won't object to this if the intent was to provide clarity.